aws-s3/authorization.ml

```diff
@@ -126,13 +126,17 @@ let make_presigned_url ?(scheme=`Https) ?host ?port ~credentials ~date ~region ~
   let duration = string_of_int duration in
   let region = Region.to_string region in
   let headers = Headers.singleton "Host" host_header in
-  let query = [
```

Comment:
I think this would be easier to read, if the `partial_query` was in the scope of query:

```ocaml
let query = 
  let base =  [
        ("X-Amz-Algorithm", "AWS4-HMAC-SHA256");
        ("X-Amz-Credential", sprintf "%s/%s/%s/s3/aws4_request" credentials.Credentials.access_key date region);
        ("X-Amz-Date", sprintf "%sT%sZ" date time);
        ("X-Amz-Expires", duration);
        ("X-Amz-SignedHeaders", "host");
      ] 
  in
  match credentials.token with
    | Some t -> ("X-Amz-Security-Token", t) :: partial_query
    | None -> partial_query
in
```